test/blackbox/run_blackbox_tests/test_cases_submesh02.c

```diff
@@ -0,0 +1,248 @@
+static char *test_case_submesh_2_nodes[] = { "corenode1", "app1node1", "app2node1", "corenode2", "app1node2", "app2node2" };
+static black_box_state_t test_case_submesh_2_state = {
+	.test_case_name =  "test_cases_submesh02",
+	.node_names =  test_case_submesh_2_nodes,
+	.num_nodes =  6
+};
+
+static int black_box_group0_setup(void **state) {
+	const char *nodes[] = { "corenode1", "app1node1", "app2node1", "corenode2", "app1node2", "app2node2" };
+	int num_nodes = sizeof(nodes) / sizeof(nodes[0]);
+
+	PRINT_TEST_CASE_MSG("Creating Containers\n");
+	destroy_containers();
+	create_containers(nodes, num_nodes);
+
+	return 0;
+}
+
+static int black_box_group0_teardown(void **state) {
+	PRINT_TEST_CASE_MSG("Destroying Containers\n");
+	destroy_containers();
+
+	return 0;
+}
+
+static int black_box_all_nodes_setup(void **state) {
+	const char *nodes[] = { "corenode1" };
+	int num_nodes = sizeof(nodes) / sizeof(nodes[0]);
+
+	PRINT_TEST_CASE_MSG("Creating Containers\n");
+	destroy_containers();
+	create_containers(nodes, num_nodes);
+	PRINT_TEST_CASE_MSG("Created Containers\n");
+	return 0;
+}
+
+static void signal_node_start(node_status_t *node_status, int start,
+                              int end) {
+	int i;
+
+	for(i = start; i <= end; i++) {
+		if(NODE_JOINED != node_status[i].currentEv) {
+			return;
+		}
+	}
+
+	for(i = start; i <= end; i++) {
+		node_step_in_container(node_ids[i], "SIGIO");
+	}
+
+	return;
+}
+
+static bool event_cb(mesh_event_payload_t payload) {
+	static node_status_t node_status[6] = { {INIT_ST, core_node1, 0, 3, false},
+		{INIT_ST, app1_node1, 0, 4, false},
+		{INIT_ST, app2_node1, 0, 4, false},
+		{INIT_ST, core_node2, 0, 4, false},
+		{INIT_ST, app1_node2, 0, 7, false},
+		{INIT_ST, app2_node2, 0, 7, false}
+	};
+
+	fprintf(stderr, "%s(%lu) : ", event_node_name[payload.client_id], time(NULL));
+
+	switch(payload.mesh_event) {
+	case NODE_STARTED:
+		fprintf(stderr, "Node Started\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case NODE_JOINED:
+		fprintf(stderr, "Node Joined\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		signal_node_start(node_status, 1, 5);
+		break;
+
+	case ERR_NETWORK:
+		fprintf(stderr, "Network Error\n");
+		assert(false);
+		break;
+
+	case CHANNEL_OPENED:
+		fprintf(stderr, "Channel Opened\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case CHANNEL_DATA_RECIEVED:
+		fprintf(stderr, "Channel Data Received\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case SIG_ABORT:
+		fprintf(stderr, "SIG_ABORT Received\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case MESH_EVENT_COMPLETED:
+		fprintf(stderr, "MESH_EVENT_COMPLETED Received\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+	}
```

Comment:
Almost all of the cases in this switch-statement effectively do the same thing. If you don't care too much about the print statements, it can be simplified to:
```
fprintf(stderr, "%s(%lu) : Received event %u", event_node_name[payload.client_id], time(NULL), payload.mesh_event);
assert(change_state(&node_status[payload.client_id], payload.mesh_event)); // Will handle ERR_NETWORK correctly

if(payload.mesh_event == NODE_JOINED) {
        signal_node_start(node_status, 1, 5);
}
```